gp-includes/things/translation.php

```diff
@@ -415,15 +415,54 @@ public function for_translation( $project, $translation_set, $page, $filters = a
 
 		$sql_sort = sprintf( $sort_by, $sort_how );
 
-		$limit = $this->sql_limit_for_paging( $page, $this->per_page );
+		/**
+		 * Filter the translation sets to allow falling back to another translation set.
+		 *
+		 * @since 2.2.0
+		 *
+		 * @param null Default for no additional translation set.
+		 * @param GP_Translation_Set $translation_set Current translation set.
+		 */
+		$fallback_translation_set = apply_filters( 'gp_for_translation_fallback_translation_set', null, $translation_set );
+
+		$translation_sets = array( $translation_set->id );
+
+		if ( $fallback_translation_set ) {
+			$limit = $this->sql_limit_for_paging_with_fallback( $page, $this->per_page );
+			$translation_sets[] = (int) $fallback_translation_set;
+		} else {
+			$limit = $this->sql_limit_for_paging( $page, $this->per_page );
+		}
+
+		$translation_sets = implode( ',', $translation_sets );
 
 		$sql_for_translations = "
 			SELECT SQL_CALC_FOUND_ROWS t.*, o.*, t.id as id, o.id as original_id, t.status as translation_status, o.status as original_status, t.date_added as translation_added, o.date_added as original_added
 			FROM $wpdb->gp_originals as o
-			$join_type JOIN $wpdb->gp_translations AS t ON o.id = t.original_id AND t.translation_set_id = " . (int) $translation_set->id . " $join_where
+			$join_type JOIN $wpdb->gp_translations AS t ON o.id = t.original_id AND t.translation_set_id IN( " . $wpdb->esc_like( $translation_sets ) . " ) $join_where
```

Comment:
Because of the above I don't really see an other option (even if it's ugly because of the field naming) than making this two joins for variants and using `COALESCE()` to get the appropriate column. Something like:
```
		$sql_for_translations = "
			SELECT SQL_CALC_FOUND_ROWS COALESCE(t_var.id, t.id), COALESCE(t_var.original_id, t.original_id), COALESCE(t_var.translation_set_id, t.translation_set_id), COALESCE(t_var.translation_0, t.translation_0), COALESCE(t_var.translation_1, t.translation_1), COALESCE(t_var.translation_2, t.translation_2), COALESCE(t_var.translation_3, t.translation_3), COALESCE(t_var.translation_4, t.translation_4), COALESCE(t_var.translation_5, t.translation_5), COALESCE(t_var.user_id, t.user_id), COALESCE(t_var.status, t.status), COALESCE(t_var.date_added, t.date_added), COALESCE(t_var.date_modified, t.date_modified), COALESCE(t_var.warnings, t.warnings), COALESCE(t_var.user_id_last_modified, t.user_id_last_modified), o.*, COALESCE(t_var.id, t.id) as id, o.id as original_id, COALESCE(t_var.status, t.status) as translation_status, o.status as original_status, COALESCE(t_var.date_added, t.date_added) as translation_added, o.date_added as original_added
			FROM $wpdb->gp_originals as o
			$join_type JOIN $wpdb->gp_translations AS t ON o.id = t.original_id AND t.translation_set_id = " . (int) $translation_set->id . " 
			$join_type JOIN $wpdb->gp_translations AS t_var ON o.id = t_var.original_id AND t_var.translation_set_id = " . intval( $fallback_translation_set ) . " $join_where
			WHERE o.project_id = " . (int) $project->id . " AND o.status = '+active' $where ORDER BY $sql_sort $limit";
```